roles/common/tasks/secrets.yml

```diff
@@ -16,3 +16,15 @@
   when: |
     inventory_hostname in groups.repository | default([]) +  groups.search | default([])
     and hostvars.localhost.reposearch_shared_secret is defined
+
+- name: Set external activemq credentials if present
+  set_fact:
+    activemq_username: "{{ groups.external_activemq | map('extract', hostvars, ['activemq_username']) | first }}"
+    activemq_password: "{{ groups.external_activemq | map('extract', hostvars, ['activemq_password']) | first }}"
+  when: groups.external_activemq | default(False)
+
+- name: Fallback to default insecure credentials without external activemq
+  set_fact:
+    activemq_username: admin
+    activemq_password: admin
```

Comment:
Let's do it like that for now, but that's the reason why I think we need a proper separate secret playbook we just import rather than something in the common role. That way we can make default password generation happen outside of the main playbook leaving room for ppl to amend generated password if they want to and don't need to rely on default stuff